src/Plugin/RulesDataMatcher/MixedGreaterThanMatcher.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Plugin\RulesDataMatcher\MixedGreaterThanMatcher.
+ */
+
+namespace Drupal\rules\Plugin\RulesDataMatcher;
+
+use Drupal\rules\Context\DataMatcherInterface;
+
+/**
+ * Defines a 'greater than' matcher.
+ *
+ * @RulesDataMatcher(
+ *   id = "rules_mixed_greater_than",
+ *   label = @Translation("A 'greater than' matcher."),
+ *   subject_type = "mixed",
```

Comment:
I would say it would be better to pass two value objects implementing a ComparableInterface as arguments and implement it as something like this:

// throws NotComparableException if the two can't be compared
return $subject->compareTo($object);